test/hashing.spec.js

```diff
@@ -0,0 +1,1347 @@
+    expect(res).to.be.equal('ni:///sha-256;b19f1beb88382b7a670c276a699a61c9805174586eaeb1845c4a1dbce8b75aa3?ver=CBV2.0');
+  });
+});
+
+describe('dl normalizer', () => {
+  it('Should return a valid gtin', () => {
+    expect(addCheckDigitAndZeroPad('7447010150', true)).to.be.equal('00074470101505');
+    expect(addCheckDigitAndZeroPad('7447010150', true, 13)).to.be.equal('0074470101505');
+    expect(addCheckDigitAndZeroPad('7447010150', true, 12)).to.be.equal('074470101505');
+  });
+
+  it('Should return the input since it is not a normalisable uri', () => {
+    expect(dlNormalizer('foo')).to.be.equal('foo');
+    expect(dlNormalizer('')).to.be.equal('');
+  });
+
+  it('Should normalize an urn with sgtin', () => {
+    expect(dlNormalizer('urn:epc:id:sgtin:0614141.011111.987')).to.be.equal('https://id.gs1.org/01/00614141111114/21/987');
+  });
+
+  it('Should normalize an urn with sscc', () => {
+    expect(dlNormalizer('urn:epc:id:sscc:4047023.0111111122')).to.be.equal('https://id.gs1.org/00/0404702301111111225');
+  });
+
+  it('Should normalize an urn with sgln', () => {
+    expect(dlNormalizer('urn:epc:id:sgln:4012345.00012.0')).to.be.equal('https://id.gs1.org/414/4012345000122');
+    expect(dlNormalizer('urn:epc:id:sgln:0012345.11111.0')).to.be.equal('https://id.gs1.org/414/0012345111112');
+    expect(dlNormalizer('urn:epc:id:sgln:4012345.00011.0')).to.be.equal('https://id.gs1.org/414/4012345000115');
+  });
+
+  it('Should normalize an urn with grai', () => {
+    expect(dlNormalizer('urn:epc:id:grai:4012345.00022.334455')).to.be.equal('https://id.gs1.org/8003/04012345000221334455');
+  });
+
+  it('Should normalize an urn with giai', () => {
+    expect(dlNormalizer('urn:epc:id:giai:4000001.111')).to.be.equal('https://id.gs1.org/8004/4000001111');
+  });
+
+  it('Should normalize an urn with gsrn', () => {
+    expect(dlNormalizer('urn:epc:id:gsrn:4012345.0000006765')).to.be.equal('https://id.gs1.org/8018/401234500000067657');
+  });
+
+  it('Should normalize an urn with gsrnp', () => {
+    expect(dlNormalizer('urn:epc:id:gsrnp:4012345.0000000007')).to.be.equal('https://id.gs1.org/8017/401234500000000074');
+  });
+
+  it('Should normalize an urn with gdti', () => {
+    expect(dlNormalizer('urn:epc:id:gdti:4012345.55555.1234')).to.be.equal('https://id.gs1.org/253/40123455555541234');
+  });
+
+  it('Should normalize an urn with cpi', () => {
+    expect(dlNormalizer('urn:epc:id:cpi:0614141.11111111111111-A%23%2F.1234')).to.be.equal('https://id.gs1.org/8010/061414111111111111111-A%23%2F/8011/1234');
+  });
+
+  it('Should normalize an urn with sgcn');
```

Comment:
Will this be implemented?

Reply:
I couldn't find any urn with sgcn! But I will probably find an example one day.